x-pack/plugins/ml/public/application/components/annotations/annotation_flyout/index.test.tsx

```diff
@@ -5,58 +5,96 @@
  */
 
 import useObservable from 'react-use/lib/useObservable';
-
 import mockAnnotations from '../annotations_table/__mocks__/mock_annotations.json';
-
 import React from 'react';
-import { mountWithIntl, shallowWithIntl } from '@kbn/test/jest';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
 
 import { Annotation } from '../../../../../common/types/annotations';
-import { annotation$ } from '../../../services/annotations_service';
+import { AnnotationUpdatesService } from '../../../services/annotations_service';
 
 import { AnnotationFlyout } from './index';
+import { MlAnnotationUpdatesContext } from '../../../contexts/ml/use_ml_annotation_updates';
 
-describe('AnnotationFlyout', () => {
-  test('Initialization.', () => {
-    const wrapper = shallowWithIntl(<AnnotationFlyout />);
-    expect(wrapper).toMatchSnapshot();
-  });
+jest.mock('../../../util/dependency_cache', () => ({
+  getToastNotifications: () => ({ addSuccess: jest.fn(), addDanger: jest.fn() }),
+}));
 
-  test('Update button is disabled with empty annotation', () => {
+const annotationUpdatesService = new AnnotationUpdatesService();
```

Comment:
you should instantiate a service in `before` hook

Reply:
@qn895  inside of `describe('AnnotationFlyout', () => {`
creating the following 
```typescript
let annotationUpdatesService: AnnotationUpdatesService = null

beforeEach(() => {
  annotationUpdatesService = new AnnotationUpdatesService();
})
```
to make sure each test is isolated

Reply:
My apologies - missed this comment yesterday. Updated here: https://github.com/elastic/kibana/pull/83216/commits/9b8020ba08b682b8fa5c9601c3858bdb5ad529aa